uber/models/attendee.py

```diff
@@ -671,42 +676,59 @@ def is_transferable(self):
             not self.has_role_somewhere
 
     @property
-    def paid_for_a_swag_shirt(self):
+    def paid_for_a_shirt(self):
         return self.amount_extra >= c.SHIRT_LEVEL
 
     @property
-    def volunteer_swag_shirt_eligible(self):
+    def volunteer_event_shirt_eligible(self):
         """
-        Returns: True if this attendee is eligible for a swag shirt *due to
-            their status as a volunteer or staff*. They may additionally be
-            eligible for a swag shirt for other reasons too.
+        Returns a truthy value if this attendee either automatically gets a
+        complementary event shirt for being staff OR if they've eligible for a
+        complementary event shirt if they end up working enough volunteer hours
         """
-
-        # Some events want to exclude staff badges from getting swag shirts
+        # Some events want to exclude staff badges from getting event shirts
         # (typically because they are getting staff uniform shirts instead).
         if self.badge_type == c.STAFF_BADGE:
             return c.STAFF_ELIGIBLE_FOR_SWAG_SHIRT
         else:
             return c.VOLUNTEER_RIBBON in self.ribbon_ints
 
     @property
-    def volunteer_swag_shirt_earned(self):
-        return self.volunteer_swag_shirt_eligible and (
+    def volunteer_event_shirt_earned(self):
+        return self.volunteer_event_shirt_eligible and (
             not self.takes_shifts or self.worked_hours >= 6)
 
     @property
-    def num_swag_shirts_owed(self):
-        swag_shirts = int(self.paid_for_a_swag_shirt)
-        volunteer_shirts = int(self.volunteer_swag_shirt_eligible)
-        return swag_shirts + volunteer_shirts
+    def replacement_staff_shirts(self):
+        """
+        Staffers can choose whether or not they want to swap out one of their
+        staff shirts for an event shirt.  By default and if the staffer opts
+        into this, we deduct 1 staff shirt from the staff shirt count and add 1
+        to the event shirt count.
+        """
+        is_replaced = self.second_shirt in [c.UNKNOWN, c.STAFF_AND_EVENT_SHIRT]
+        return 1 if self.gets_staff_shirt and is_replaced else 0
+
+    @property
+    def num_event_shirts_owed(self):
+        return sum([
+            int(self.paid_for_a_shirt),
+            int(self.volunteer_event_shirt_eligible),
+            self.replacement_staff_shirts
```

Comment:
The event shirts for volunteers and staff are mutually exclusive, so it's a little odd that they're both counted here. Right now this would work fine with all our presave adjustments to prevent staff from having volunteer ribbons -- volunteer_event_shirt_eligible and replacement_staff_shirts SHOULD never both be true -- but if that ever changes, we could find ourselves with a pretty tricky (and hard to track down) bug.